controllers/configmap_controller.go

```diff
@@ -18,45 +18,229 @@ package controllers
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+
+	"github.com/openshift/windows-machine-config-operator/pkg/cluster"
+	"github.com/openshift/windows-machine-config-operator/pkg/metrics"
+	"github.com/openshift/windows-machine-config-operator/pkg/secrets"
+)
+
+//+kubebuilder:rbac:groups="",resources=configmaps,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups="",resources=configmaps/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups="",resources=configmaps/finalizers,verbs=update
+
+const (
+	// BYOHAnnotation is an an anotation that should be applied to all Windows nodes not associated with a Machine.
+	BYOHAnnotation = "windowsmachineconfig.openshift.io/byoh"
+	// InstanceConfigMap is the name of the ConfigMap where VMs to be configured should be described.
+	// TODO: Possibly make this a singleton that WMCO creates https://issues.redhat.com/browse/WINC-612
+	InstanceConfigMap = "windows-instances"
 )
 
 // ConfigMapReconciler reconciles a ConfigMap object
 type ConfigMapReconciler struct {
-	client.Client
-	Log    logr.Logger
-	Scheme *runtime.Scheme
+	instanceReconciler
 }
 
-//+kubebuilder:rbac:groups=windowsmachineconfig.openshift.io,resources=configmaps,verbs=get;list;watch;create;update;patch;delete
-//+kubebuilder:rbac:groups=windowsmachineconfig.openshift.io,resources=configmaps/status,verbs=get;update;patch
-//+kubebuilder:rbac:groups=windowsmachineconfig.openshift.io,resources=configmaps/finalizers,verbs=update
+// NewConfigMapReconciler returns a pointer to a ConfigMapReconciler
+func NewConfigMapReconciler(mgr manager.Manager, clusterConfig cluster.Config, watchNamespace string) (*ConfigMapReconciler, error) {
+	clientset, err := kubernetes.NewForConfig(mgr.GetConfig())
+	if err != nil {
+		return nil, errors.Wrap(err, "error creating kubernetes clientset")
+	}
+
+	// Initialize prometheus configuration
+	pc, err := metrics.NewPrometheusNodeConfig(clientset, watchNamespace)
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to initialize Prometheus configuration")
+	}
+	return &ConfigMapReconciler{
+		instanceReconciler: instanceReconciler{
+			client:               mgr.GetClient(),
+			k8sclientset:         clientset,
+			clusterServiceCIDR:   clusterConfig.Network().GetServiceCIDR(),
+			log:                  ctrl.Log.WithName("controllers").WithName("ConfigMap"),
+			watchNamespace:       watchNamespace,
+			recorder:             mgr.GetEventRecorderFor("configmap"),
+			vxlanPort:            clusterConfig.Network().VXLANPort(),
+			prometheusNodeConfig: pc,
+		},
+	}, nil
+}
 
 // Reconcile is part of the main kubernetes reconciliation loop which aims to
 // move the current state of the cluster closer to the desired state.
-// TODO(user): Modify the Reconcile function to compare the state specified by
-// the ConfigMap object against the actual cluster state, and then
-// perform operations to make the cluster state reflect the state specified by
-// the user.
-//
```

Comment:
Is this being addressed in this PR?

Reply:
Yes it is, that todo is created as part of the generated file. This PR takes the expected windows-instances and creates nodes out of them. The missing parts of this process were moved into other TODOs